data_storage/export_format.md

```diff
@@ -1,22 +1,28 @@
 Export Format
 -------------
 
-- **5.1.1** ![](/badge/req.png) A Toxcore data file saved in the root of the client's
-  working directory, with the file name `<profile_name>.tox`
+- **5.1.1** ![](/badge/req.png) The client will export files relative to a folder called
+  the export folder, which must be in a location accessible to the user.
 
-- **5.1.2** ![](/badge/req.png) If the client supports user avatars, they will be stored
-  under the `avatars/` directory in the root of the client's working directory.
+- **5.1.2** ![](/badge/rec.png) The user should be able to choose the location
+  of the export folder.
 
-- **5.1.3** ![](/badge/req.png) Avatars are named according to the following format:
+- **5.1.3** ![](/badge/req.png) A Toxcore data file saved in the root of the
+  export folder, with the file name `<profile_name>.tox`.
```

Comment:
What is a toxcore data file? If it's the data produced by `tox_get_savedata`, I disagree, unless that format is clearly specified, and then I would not be happy with that, either, because it's a custom binary format.

I would suggest using a formally specified backwards-compatible export format for which each client can generate parsers and generators. Protobuf is an obvious choice here, as it natively supports binary and text formats (I suggest .pb and .textpb file extensions, perhaps .tox.pb/.tox.textpb to disambiguate). As of [proto3](https://github.com/google/protobuf/releases/tag/v3.0.0-alpha-2), a native json format is also [specified](https://developers.google.com/protocol-buffers/docs/proto3#json), so we could trivially support those three formats (+.tox.json?).

If I'm not mistaken, all the information stored in Tox save-data can be reconstructed from the environment or from external inputs (other API calls). Where is the save data format specified?